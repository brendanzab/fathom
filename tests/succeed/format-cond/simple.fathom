let format = { sfnt_version <- { version <- u32be | (u32_eq version 0xffff) } };

let _ : format.Repr -> { sfnt_version : u32be.Repr } =
    fun x => x;

{}
