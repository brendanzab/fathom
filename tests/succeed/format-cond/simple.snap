stdout = '''
let format : Format = {
    sfnt_version <- { version <- u32be | version == (0xffff : U32) },
};
let _ : format.Repr -> { sfnt_version : u32be.Repr } = fun x => x;
() : ()
'''
stderr = ''
